crates/biome_js_analyze/src/semantic_analyzers/nursery/use_sorted_classes/any_class_string_like.rs

```diff
@@ -0,0 +1,227 @@
+use biome_analyze::{
+    AddVisitor, AnalyzerOptions, Phases, QueryMatch, Queryable, RuleKey, ServiceBag, Visitor,
+    VisitorContext,
+};
+use biome_js_syntax::{
+    JsCallArguments, JsCallExpression, JsLanguage, JsStringLiteralExpression,
+    JsTemplateChunkElement, JsxAttribute, JsxString,
+};
+use biome_rowan::{
+    declare_node_union, AstNode, Language, SyntaxNode, TextRange, TokenText, WalkEvent,
+};
+
+use super::UseSortedClassesOptions;
+
+// utils
+// -----
+
+fn get_options_from_analyzer(analyzer_options: &AnalyzerOptions) -> UseSortedClassesOptions {
+    analyzer_options
+        .configuration
+        .rules
+        .get_rule_options::<UseSortedClassesOptions>(&RuleKey::new("nursery", "useSortedClasses"))
+        .cloned()
+        .unwrap_or_default()
+}
+
+fn get_callee_name(call_expression: &JsCallExpression) -> Option<TokenText> {
+    call_expression
+        .callee()
+        .ok()?
+        .as_js_identifier_expression()?
+        .name()
+        .ok()?
+        .name()
+        .ok()
+}
+
+fn is_call_expression_of_target_function(
+    call_expression: &JsCallExpression,
+    target_functions: &[String],
+) -> bool {
+    get_callee_name(call_expression)
+        .is_some_and(|name| target_functions.contains(&name.to_string()))
+}
+
+fn get_attribute_name(attribute: &JsxAttribute) -> Option<TokenText> {
+    Some(
+        attribute
+            .name()
+            .ok()?
+            .as_jsx_name()?
+            .value_token()
+            .ok()?
+            .token_text_trimmed(),
+    )
+}
+
+fn is_target_attribute(attribute: &JsxAttribute, target_attributes: &[String]) -> bool {
+    match get_attribute_name(attribute) {
+        Some(name) => target_attributes.contains(&name.to_string()),
+        None => false,
+    }
+}
+
+// attributes visitor
+// ------------------
+
+#[derive(Default)]
+struct StringLiteralInAttributeVisitor {
+    in_target_attribute: bool,
+}
+
+// Finds class-like strings in JSX attributes, including class, className, and others defined in the options.
+impl Visitor for StringLiteralInAttributeVisitor {
+    type Language = JsLanguage;
+    fn visit(
+        &mut self,
+        event: &WalkEvent<SyntaxNode<Self::Language>>,
+        mut ctx: VisitorContext<Self::Language>,
+    ) {
+        let options = get_options_from_analyzer(ctx.options);
```

Comment:
Yeah that’s definitely a good guess!

Reply:
yeah this is one of my main suspects, though I'm not sure how I could cache it. maybe the queryable trait could have a method to pre-load some stuff that is then made available through `ctx`.

Reply:
The analyzer already has a system in place for this. They are called services, they are cached among runs. Other rules use this system. Check the "top level hooks" rule for examples.

I'll look into it in case you can't.